db/seeds/db_populator.rb

```diff
@@ -0,0 +1,151 @@
+# Called by the seeding process to create data with a specified random number generator.
+# There is a 1 in 30 probability that a Volunteer will be inactive when created.
+# There is no instance of a volunteer who was previously assigned a case being inactivated.
+# Email addresses generated will be globally unique across all orgs.
+
+class DbPopulator
+
+  SEED_PASSWORD   = "123456"
+
+  attr_reader :rng
+
+  # Public Methods
+
+  # Pass an instance of Random for Faker and Ruby `rand` and sample` calls.
+  def initialize(random_instance)
+    @rng = random_instance
+    @casa_org_counter = 0
+    @case_number_sequence = 1000
+  end
+
+  def create_all_casa_admin(email = 'allcasaadmin@example.com')
+    AllCasaAdmin.create!(email: email,  password: SEED_PASSWORD, password_confirmation: SEED_PASSWORD)
+  end
+
+  # See CasaOrgPopulatorPresets for the content of the options hash.
+  def create_org(options_hash)
+
+    options = OpenStruct.new(options_hash)
+    @casa_org_counter += 1
+
+    options.org_name ||= "CASA Organization ##{@casa_org_counter}"
+    casa_org = CasaOrg.find_or_create_by!(name: options.org_name) do |org|
+      org.name = options.org_name
+      org.display_name = options.org_name
+      org.casa_org_logo = logo
+      org.address = Faker::Address.full_address
+      org.footer_links = [
+          ["https://example.org/contact/", "Contact Us"],
+          ["https://example.org/subscribe-to-newsletter/", "Subscribe to newsletter"],
+          ["https://www.example.org/give/givefrm.asp?CID=4450", "Donate"]
+      ]
+    end
+
+    create_users(casa_org, options)
+    create_cases(casa_org, options)
+    casa_org
+  end
+
+  private # -------------------------------------------------------------------------------------------------------
+
+  # Creates 3 users, 1 each for [Volunteer, Supervisor, CasaAdmin].
+  # For org's after the first one created, adds an org number to the email address so that they will be globally unique
+  def create_users(casa_org, options)
+
+    # Generate email address; for orgs only after first org, and org number would be added, e.g.:
+    # Org #1: volunteer1@example.com
+    # Org #2: volunteer2-1@example.com
+    email = ->(klass, n) do
+      org_fragment = (@casa_org_counter > 1) ? "#{@casa_org_counter}-" : ''
+      klass.name.underscore + org_fragment + n.to_s + '@example.com'
+    end
+
+    create_users_of_type = ->(klass, count) do
+      (1..count).each do |n|
+        attributes = {
+            casa_org: casa_org,
+            email: email.(klass, n),
+            display_name: Faker::Name.name,
+            password: SEED_PASSWORD,
+            password_confirmation: SEED_PASSWORD
+        }
+        if klass == 'Volunteer' && random(30, random: rng) == 0
```

Comment:
Can we just use mod here to ensure that at least every 30th record is marked as inactive instead of relying on the randomization?

```
if klass == 'Volunteer' && n % 30 == 0
```

Reply:
Yes, as long as you're ok with the lesser amount of variation. For example, if we test against mod zero then specifying a single record for the seeding would result in that sole record always being inactive, and that might be confusing; and there are many more examples. Now that we have the option to specify a random seed we can have or not have predictability as we like.

Reply:
Okay, either is fine! Welcome to leave it as is.